electrum/wallet.py

```diff
@@ -263,6 +263,17 @@ def save_db(self):
         if self.storage:
             self.db.write(self.storage)
 
+    def save_backup(self, path, *, old_password=None, new_password=None):
+        new_db = WalletDB(self.storage.read(), manual_upgrades=False)
```

Comment:
could `self.storage.read()` return an older state of the db than `db` (which is current)?
does it not simply return the state when the storage object was created (which might be when the wallet was opened)?

Reply:
right, thats a very good point.